pkg/controller/vpc_nat_gw_nat.go

```diff
@@ -100,9 +101,21 @@ func (c *Controller) enqueueUpdateIptablesDnatRule(old, new interface{}) {
 
 	if oldDnat.Status.V4ip != newDnat.Status.V4ip ||
 		oldDnat.Spec.EIP != newDnat.Spec.EIP ||
-		oldDnat.Status.Redo != newDnat.Status.Redo {
+		oldDnat.Status.Redo != newDnat.Status.Redo ||
+		oldDnat.Spec.Protocol != newDnat.Spec.Protocol ||
+		oldDnat.Spec.InternalIp != newDnat.Spec.InternalIp ||
+		oldDnat.Spec.InternalPort != newDnat.Spec.InternalPort ||
+		oldDnat.Spec.ExternalPort != newDnat.Spec.ExternalPort {
 		klog.V(3).Infof("enqueue update dnat %s", key)
-		c.updateIptablesDnatRuleQueue.Add(key)
+		raw, _ := json.Marshal(map[string]string{
```

Comment:
最好还是把 key 加入队列，然后在 handle 的时候直接把所有规则 flush，或者差量更新。

Reply:
I've changed to store the information from the old version to status, so that when updating the rule, I could obtain last status information to delete the obsolete rules.